LiskHQ/lisk-hub
# Improve logic in hw manager - closes #2418

### What issue have I solved?

-- #2418 

### How have I implemented/fixed it?

This PR is just for improve the logic in the HWManager and remove the executeCommand from trezor.js and ledger.js file and use a specific function for each action.

- [x] Remove executeCommand function from trezor and ledger files.
- [x] Created function for getPublicKey in trezor and ledger files.
- [x] Created function for signTransactions in trezor and ledger files.
- [x] Add send ipc message function for connected device. (for toaster message in react)
- [x] Add send ipc message function for disconnected device. (for toaster message in react).


### How has this been tested?

1. Connect . the trezor or ledger device.
2. login with any of this devices.
3. vote for a delegate or unvote
4. disconnect the device
5. app should log you out.

All this should be happening without any error, and you should be able to see the toaster messages for connect and disconnect devices.

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)
